sky/provision/docker_utils.py

```diff
@@ -224,12 +224,26 @@ def initialize(self) -> str:
             # TODO(tian): Maybe support a command to get the login password?
             docker_login_config = DockerLoginConfig(
                 **self.docker_config['docker_login_config'])
-            self._run(
-                f'{self.docker_cmd} login --username '
-                f'{docker_login_config.username} '
-                f'--password {docker_login_config.password} '
-                f'{docker_login_config.server}',
-                wait_for_docker_daemon=True)
+            if docker_login_config.password:
+                # Password is allowed to be empty, in that case, we will not run
+                # the login command, and assume that the image pulling is
+                # authenticated by the IAM permission on the VM.
+                self._run(
+                    f'{self.docker_cmd} login --username '
+                    f'{shlex.quote(docker_login_config.username)} '
+                    f'--password {shlex.quote(docker_login_config.password)} '
+                    f'{shlex.quote(docker_login_config.server)}',
+                    wait_for_docker_daemon=True)
+            elif docker_login_config.server.endswith('-docker.pkg.dev'):
+                # Docker image server is on GCR, we need to do additional setup
+                # to pull the image.
+                # When no username or password is provided, we assume that
+                # we are on GCP VM (i.e. gcloud auth configure-docker is
+                # enough), or the image server is public.
+                # For the former case, gcloud should be available, and latter
+                # should be fine to fail the following command.
```

Comment:
may have a confusing error in this case, otherwise makes sense

Reply:
It might be fine as it is in provision.log